grails-app/views/order/_itemDetails.gsp

```diff
@@ -1,80 +1,122 @@
-                <g:set var="isItemCanceled" value="${orderItem.orderItemStatusCode == OrderItemStatusCode.CANCELED}"/>
-                <g:if test="${!isItemCanceled || orderInstance?.orderType==OrderType.findByCode(OrderTypeCode.PURCHASE_ORDER.name())}">
-                    <tr class="order-item ${(i % 2) == 0 ? 'even' : 'odd'}" style="${isItemCanceled ? 'background-color: #ffcccb;' : ''}">
-                        <td>
-                            ${orderItem?.product?.productCode?:""}
-                        </td>
-                        <td class="order-item-product">
-                            <g:link controller="inventoryItem" action="showStockCard" params="['product.id':orderItem?.product?.id]">
-                                <format:product product="${orderItem?.product}"/>
-                                <g:renderHandlingIcons product="${orderItem?.product}" />
-                            </g:link>
-                        </td>
-                        <g:if test="${!isItemCanceled}">
-                            <td class="center">
-                                ${orderItem?.productSupplier?.supplierCode}
-                            </td>
-                            <td class="center">
-                                ${orderItem?.productSupplier?.manufacturerName}
-                            </td>
-                            <td class="center">
-                                ${orderItem?.productSupplier?.manufacturerCode}
-                            </td>
-                            <td class="center">
-                                ${orderItem?.quantity }
-                            </td>
-                            <td class="center">
-                                ${orderItem?.unitOfMeasure}
-                            </td>
-                            <td class="center">
-                                ${orderItem?.recipient}
-                            </td>
-                            <td class="center">
-                                <g:formatDate date="${orderItem?.estimatedReadyDate}" format="dd/MMM/yyyy"/>
-                            </td>
-                            <td class="center">
-                                <g:formatDate date="${orderItem?.actualReadyDate}" format="dd/MMM/yyyy"/>
+<%@ page import="org.pih.warehouse.core.Constants;" %>
+
+
+<script>
+  $(document).ready(function() {
+    $("#orderItemsDetailsFilter").keyup(function(event){
+      const filterCell = 1; // product name
+      const filterValue = $("#orderItemsDetailsFilter")
+        .val()
+        .toUpperCase();
+      filterTableItemDetails(filterCell, filterValue)
+    });
+
+  });
+  function filterTableItemDetails(cellIndex, filterValue) {
+    const tableRows = $("#order-items-details tr.dataRowItemDetails");
+    // Loop through all table rows, and hide those who don't match the search query
+    $.each(tableRows, function(index, currentRow) {
```

Comment:
Pass the `tableRows` to this function, and extract the rest, because it is common for all 3 filter functions.